torch/ao/quantization/pt2e/representation/rewrite.py

```diff
@@ -95,6 +98,41 @@ def _reference_quantized_add(
     out_i8 = torch.ops.aten.clamp(out_i32, quant_min, quant_max).to(torch.int8)
     return out_i8
 
+_QUANTIZED_MAX_POOL2D_EXAMPLE_INPUTS = (
+    torch.randint(-128, 127, (1, 3, 3, 3), dtype=torch.int8),
+    torch.randn(1, dtype=torch.float),
+    torch.zeros(1, dtype=torch.int),
+    torch.randn(1, dtype=torch.float),
+    torch.zeros(1, dtype=torch.int),
+    torch.tensor([-128], dtype=torch.int),
+    torch.tensor([127], dtype=torch.int),
+)
+
+def _qdq_quantized_max_pool2d(x_i8, x_scale, x_zero_point, out_scale, out_zero_point, quant_min, quant_max):
+    kernel_size = 1
+    stride = 1
+    padding = 0
+    dilation = 1
+    ceil_mode = False
+    x_fp32 = torch.ops.quantized_decomposed.dequantize_per_tensor(x_i8, x_scale, x_zero_point, quant_min, quant_max, torch.int8)
+    out_fp32, _ = torch.ops.aten.max_pool2d_with_indices.default(x_fp32, kernel_size, stride, padding, dilation, ceil_mode)
+    out_i8 = torch.ops.quantized_decomposed.quantize_per_tensor(out_fp32, out_scale, out_zero_point, quant_min, quant_max, torch.int8)
+    return out_i8
+
+def _reference_quantized_max_pool2d(x_i8, x_scale, x_zero_point, out_scale, out_zero_point, quant_min, quant_max):
+    kernel_size = 1
+    stride = 1
+    padding = 0
+    dilation = 1
+    ceil_mode = False
+    acc_i8, _ = torch.ops.aten.max_pool2d_with_indices.default(x_i8, kernel_size, stride, padding, dilation, ceil_mode)
+    acc_i32 = acc_i8.to(torch.int32)
+    # TODO: use mul.Scalar, need to change how we handle literal args
+    output_i32 = out_dtype(torch.ops.aten.mul.Tensor, torch.int32, acc_i32, (x_scale / out_scale))
+    output_i8 = output_i32 - (x_zero_point * x_scale / out_scale + out_zero_point)
```

Comment:
Also I dont think `(x_zero_point * x_scale / out_scale + out_zero_point)` is correct, right? affine quant is
`fp32 = scale * (int8 - zero_point)` so `int8 = fp32/scale + zero_point)`?

Has this been functionally tested? If not we should add functional tests for each.

Reply:
probably haven't test this one locally, let me check after changing to int16

Reply:
I wasnt pointing out the int16 issue here btw. I was pointing out a potential but because in the expression above you are subtracting out_zero_point, but you actually need to add

Reply:
yeah understood, I'll update the representation with int32 maxpool and make sure to check the numerics locally